ios/FluentUI/Avatar/Avatar.swift

```diff
@@ -100,46 +108,63 @@ public struct Avatar: View, TokenizedControlView {
 
     public var body: some View {
         let style = state.style
+        let size = state.size
+
+        let initialsString: String = ((style == .overflow) ? state.primaryText ?? "" : Avatar.initialsText(fromPrimaryText: state.primaryText,
+                                                                                                           secondaryText: state.secondaryText))
+        let shouldUseCalculatedColors = !initialsString.isEmpty && style != .overflow
+        let shouldUseDefaultImage = (state.image == nil && initialsString.isEmpty && style != .overflow)
+
+        let activityStyle = state.activityStyle
         let presence = state.presence
-        let shouldDisplayPresence = presence != .none
+        let shouldDisplayActivity = state.activityImage != nil && activityStyle != .none && (style == .default && !shouldUseDefaultImage)
```

Comment:
If the size of the presence or activity is zero, we should probably just say we shouldn't show them, rather than doing all the work to show something with zero size.

Reply:
Seconded - this is more directly getting at the comment I made earlier.

Reply:
Added additional checks!